scripts/pdpModule.js

```diff
@@ -0,0 +1,140 @@
+var pdpModule = (function () {
+    const categoriesItemsList = fetchCategoriesList();
+    const menuItemsList = fetchMenuList();
+    const cartItemsList = fetchCartList();
+    const itemsByCategoryMap = createItemsByCategoryMap(menuItemsList);
+
+    return {
+        init: display
+    };
+
+    function display() {
+        _displayCategories("recommended");
+        _displayMenu("recommended")
+        _displayCart();
+    }
+
+    function _displayCategories(category) {
+        const categoriesContainer = document.querySelector(".categories");
+        const categoriesItemsEl = document.createElement("ul");
+        categoriesItemsList.forEach(element => {
+            const categoriesItemEl = document.createElement("li");
+            if (element.id == category) {
+                categoriesItemEl.classList.add("highlighted");
+            }
+            categoriesItemEl.id = element.id;
+            categoriesItemEl.innerText = element.displayName;
+            categoriesItemsEl.append(categoriesItemEl);
+        });
+        categoriesContainer.append(categoriesItemsEl);
+    }
+
+    function _createMenuItem(category, index) {
+        const menuItem = getItem(itemsByCategoryMap, category, index);
```

Comment:
You're doing repetitive stuff here. 

Whenever you see such patterns, think in terms of DRY and try to extract them into a meaningful util wherever possible

An example could look like:
```

    const createDomNode = ({ type, classList, textContent }) => {
        const node = document.createElement(type);
        // if classList exists, add to node.classList
        // if textContent exists, add that
        return node;
    }
```